src/processes/atmos/atmosphere.jl

```diff
@@ -0,0 +1,123 @@
+abstract type AbstractSolarRadiation end
+
+abstract type AbstractPrecipitation end
+
+"""
+Generic type representing the concentration of a particular tracer gas in the atmosphere.
+"""
+@kwdef struct TracerGas{T}
+    "Name of the tracer gas"
+    name::Symbol
+
+    "Tracer gas concentration in ppm"
+    conc::T = nothing
+end
+
+variables(gas::TracerGas) = (
+    auxiliary(gas.name, XY(), units=u"ppm", desc="Ambient atmospheric $(gas.name) concentration in ppm"),
+)
+
+function compute_auxiliary!(state, model, gas::TracerGas)
+    conc = getproperty(state, gas.name)
+    set!(conc, gas.conc)
+end
+
+"""
+Createas a `TracerGas` for ambient CO2 with a prescribed concentration `conc`.
+"""
+AmbientCO2(conc) = TracerGas(:CO2, conc)
+
+@kwdef struct AtmosphericState{
```

Comment:
I like "AtmosphericState" the only confusing I see arising is that it's actually only the atmospheric state in one grid box, not globally? Or is your idea that `::AirTemp` could be a scalar or an array?

Reply:
No, it can be an `AbstractArray`, right? 

But we actually would want this to also be flexible enough to take a `RingGrid` here, right?